internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -75,6 +83,14 @@ func (v *TxnOp[R]) If(cs ...etcd.Cmp) *TxnOp[R] {
 // The Then operations will be executed if all If comparisons succeed.
 // To add a transaction to the Then branch, use ThenTxn, or use Merge to merge transactions.
 func (v *TxnOp[R]) Then(ops ...Op) *TxnOp[R] {
+	// Check common high-level transaction types.
+	// Bulletproof check of the low-level transaction is in the "lowLevelTxn" method.
+	for i, op := range ops {
+		if _, ok := op.(txnInterface); ok {
+			panic(errors.Errorf(`invalid operation[%d]: op is a transaction, use ThenTxn, not Then`, i))
+		}
+	}
+
```

Comment:
I added the additional check, see https://github.com/keboola/keboola-as-code/pull/1521#discussion_r1455367562

Reply:
No, because only `*TxnOp` will match it, but not your custom `AcquireLock` operation from your custom package.